### PR TITLE
[UI] add mapper to podname and version

### LINKED ISSUE #3955: [UI] Make the pod name and pod version in Control Plane Dashboard "pretty"
#### Current Behavior

The name of the control plane pods are not really pleasant. For example, `grafana-79adasdasd-aseasda`.
image

Some pod has "v" in front of its versions. Refer to pic above for non-"v" versioning pods.
image

#### Desired Behavior

There should be a way to show a "prettier" name for the pods. And also, for the version column, some version has "v" initially, some are not (refer to pic). It might be good to normalize this naming as well.

#### Implementation

Create a function to map this strings accordingly. 

#### Mockups

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by bariqhibat**
Signed-off-by: Bariq <bariqhibat@gmail.com>

**Description**

This PR fixes #3955

**Notes for Reviewers**
Making a function where all the heavy work and bad codes resides.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by warunicorn19 — CHANGES_REQUESTED** [review]
Let's keep the version as is, since it comes under a Versions column, we don't really need to add the 'v'. 
Also, if you could refer to these suggestions from nithish: https://github.com/meshery/meshery/issues/3955#issuecomment-903239452

File: ui/utils/nameMapper.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export const podNameMapper = (serviceMeshName, podName) => {
+  switch (serviceMeshName) {
+    case "istio":
+      if (podName.includes("istio-galley")) return "Istio Galley";
+      else if (podName.includes("istio-ingressgateway")) return "Istio Ingress Gateway";
```
Comment (utkarsh-pro) [inline ui/utils/nameMapper.js:5]: Minor nitpick, you can get rid of most the `else` here. Makes it more readable. :smile:

**Comment by leecalcote** [comment]
Uh-oh, a merge conflict has arisen.

File: ui/utils/nameMapper.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+export const podNameMapper = (serviceMeshName, podName) => {
```
Comment (leecalcote) [inline ui/utils/nameMapper.js:1]: @bariqhibat thank you for pulling this out separately.

File: ui/utils/nameMapper.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      break;
+    case "kiali":
+      return "Kiali";
+    case "grafana":
+      return "Grafana";
+    case "prometheus":
+      return "Prometheus";
+    case "osm":
+      if (podName.includes("osm-injector")) return "OSM Injector";
+      if (podName.includes("osm-controller")) return "OSM Controller";
```
Comment (leecalcote) [inline ui/utils/nameMapper.js:22]: If "Open Service Mesh" fits, it'd be good to expand OSM to that.

**Review by leecalcote — APPROVED** [review]
LGTM

**Pull request merged**